### DISCUSSION TITLE
How to import Altium PCBDoc

### DISCUSSION THREAD

**Discussion by danyhm** [INTI-CMNB/KiBot#625] [Q&A] [answered]
Hello,

I'm looking for a workflow where I can import Altium PCBdoc and then generate Gerber files.

Currently this is possible manually with Kicad and PCBeditor but I can't seem to find a way to automate it without UI.

Is Kibot able to do this?

**Comment by set-soft** [comment]
Hi @danyhm !

Nope, KiBot doesn't implement it. This is because this is an operation that people does only once and manually.

Are you sure that the manual process is simple? I mean: KiCad could import the file, but you'll most probably need to adjust some details manually.

**Reply by danyhm** [reply]
Hi,

The import is done with PCBeditor.

No other manual setting is required. The file is converted to kicad_pcb

**Reply by set-soft** [reply]
Do you have a small simple file to use as example? Something simple, but enough to check the gerbers

**Reply by danyhm** [reply]
[esp01.zip](https://github.com/INTI-CMNB/KiBot/files/15432256/esp01.zip)
here is a sample PCBDoc File

**Comment by danyhm** [comment]
Any chance this will be implemented?
if not do you know of any other method that doesn't need GUI and can be run in linux?
my ultimate goal is to convert PCBDoc to gerbers without altium

**Reply by set-soft** [reply]
> Any chance this will be implemented?

Not sure, it needs some user intervention, with the provided example I get:

![image](https://github.com/INTI-CMNB/KiBot/assets/22823627/8b73c549-fbf9-4703-8bcb-2894c8af1fc6)

> if not do you know of any other method that doesn't need GUI and can be run in linux?

Nope

> my ultimate goal is to convert PCBDoc to gerbers without altium

Ok, I'll take a look

**Reply by danyhm** [reply]
This is actually amazing for a start. the important layers are copper and silk layers. altium has more layers unique to its ecosystem.
How did you get this output?

**Reply by danyhm** [reply]
suppose we suppress the warnings and proceed as is. will it give any Gerbers that can be at least rendered?

**Reply by set-soft** [reply]
Not gerbers yet, but [KiAuto](https://github.com/INTI-CMNB/KiAuto) can now convert the file you sent me, I tried it for KiCad 7 and 8.

The command is: `pcbnew_do  convert -o OUTPUT.kicad_pcb INPUT.PcbDoc ./`

Now I'll try to add the conversion to KiBot

**Reply by set-soft** [reply, marked as answer]
The f0c8e12ca4987d455aeb731da49375f38e61b3de patch adds some experimental support.
You need KiAuto 2.3.2 installed, then you just pass the Altium PCB file to KiBot.
I was able to get the gerbers for the file you posted.

**Reply by danyhm** [reply]
Thanks. I see that KiAuto does this emulating the User interaction. Is there anyway to do it inside a linux VPS without GUI?

**Reply by set-soft** [reply]
KiAuto uses a virtual X server, is designed to do the work using a remote server. The main idea behind KiBot (which uses KiAuto for GUI tasks) is to run all the things in a CI/CD environment, like GitHub Actions. When running KiAuto you won't see any KiCad window, they are all redirected to a virtual server.

You can just use a docker image like [kicad_auto_test](https://github.com/INTI-CMNB/kicad_auto_test) to run KiBot. All the needed tools are installed in the docker image. But note that this new feature is experimental, and you'll need the docker images tagged as *dev* (development) to find the new features.

As I said now you can use KiBot to generate gerbers for the file you sent, you just pass to KiBot the *PcbDoc* file instead of the *kicad_pcb* file. KiBot will try to convert the file using KiAuto.

**Comment by chenzanyu** [comment]
Hi @set-soft  Using Docker KiCad 8.8, I ran the command pcbnew_do convert -o OUTPUT.kicad_pcb INPUT.PcbDoc ./, but it didn't generate the KiCad project and was instead blocked by a pop-up window. Is there any way to automatically generate it?  
```
# pcbnew_do -r convert  -o OUTPUT.kicad_pcb  esp01.PcbDoc ./
WARNING:KiCad project file not found (pcbnew_do.kiauto.file_util - file_util.py:317)
WARNING:Unknown KiCad dialog: KiCad PCB Editor Information (pcbnew_do - interposer.py:460)
WARNING:Potential dialog messages: {"Font 'Arial' not found; substituting 'Nimbus Sans Regular'.", "Font 'Calibri' not found; substituting 'DejaVu Sans Bold'.", "Font 'Arial' not found; substituting 'Nimbus Sans Bold'.", 'Details', '23:08:41', 'OK', 'H'} (pcbnew_do - interposer.py:461)
Traceback (most recent call last):
  File "/usr/bin/pcbnew_do", line 1947, in <module>
    id_pcbnew = wait_pcbnew_start_by_msg(cfg) if cfg.use_interposer else wait_pcbew_start(cfg)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/bin/pcbnew_do", line 1518, in wait_pcbnew_start_by_msg
    wait_start_by_msg(cfg)
  File "/usr/lib/python3/dist-packages/kiauto/interposer.py", line 723, in wait_start_by_msg
    res = wait_queue(cfg, pres, starts=True, timeout=cfg.wait_start, with_windows=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/kiauto/interposer.py", line 219, in wait_queue
    raise RuntimeError('Timed out waiting for `{}`'.format(strs))
RuntimeError: Timed out waiting for `['GTK:Window Title:', 'PANGO:0:']`
```

<img width="1093" alt="1745106257387" src="https://github.com/user-attachments/assets/4ac70ed9-ff9b-4000-a09a-c02cb6492567" />

**Reply by set-soft** [reply]
Can you post esp01.PcbDoc?
From the warning I suspect the issue also involves missing fonts.
If you can't post the esp01.PcbDoc post the generated video and a full log using `-vvvv`

**Reply by chenzanyu** [reply]
[esp01.zip](https://github.com/user-attachments/files/19833698/esp01.zip)
@set-soft  Of course, this is my file.

**Reply by chenzanyu** [reply]
https://github.com/INTI-CMNB/KiAuto/issues/35  this is issue

**Reply by set-soft** [reply]
Ok, lets continue it on https://github.com/INTI-CMNB/KiAuto/issues/35
